src/external/spglib.jl

```diff
@@ -1,40 +1,40 @@
-using PyCall
 # Routines for interaction with spglib
+const SPGLIB = spglib_jll.libsymspg
+
+function spglib_get_error_message()
+    error_code = ccall((:spg_get_error_code, SPGLIB), Cint, ())
+    return unsafe_string(ccall((:spg_get_error_message, SPGLIB), Cstring, (Cint,), error_code))
+end
 
 """
-Construct a tuple containing the lattice and the positions of the species
+Construct a tuple containing the positions of the species
 in the convention required to take the place of a `cell` datastructure used in spglib.
 """
-function spglib_cell_atommapping_(lattice, atoms)
-    lattice = Matrix{Float64}(lattice)  # spglib operates in double precision
-    n_attypes = isempty(atoms) ? 0 : sum(length(positions) for (type, positions) in atoms)
-    spg_numbers = Vector{Int}(undef, n_attypes)
-    spg_positions = Matrix{Float64}(undef, n_attypes, 3)
+function spglib_atommapping(atoms)
+    n_attypes = isempty(atoms) ? 0 : sum(length(positions) for (typ, positions) in atoms)
+    spg_numbers = Vector{Cint}(undef, n_attypes)
+    spg_positions = Matrix{Cdouble}(undef, 3, n_attypes)
 
     offset = 0
     nextnumber = 1
     atommapping = Dict{Int, Any}()
-    for (iatom, (type, positions)) in enumerate(atoms)
-        atommapping[nextnumber] = type
+    for (iatom, (typ, positions)) in enumerate(atoms)
+        atommapping[nextnumber] = typ
         for (ipos, pos) in enumerate(positions)
             # assign the same number to all types with this position
             spg_numbers[offset + ipos] = nextnumber
-            spg_positions[offset + ipos, :] = pos
+            spg_positions[:, offset + ipos] .= pos
         end
         offset += length(positions)
         nextnumber += 1
     end
 
-    # Note: In the python interface of spglib the lattice vectors
-    #       are given in rows, but DFTK uses columns
-    #       For future reference: The C interface spglib also uses columns.
-    (lattice', spg_positions, spg_numbers), atommapping
+    (spg_positions, spg_numbers), atommapping
 end
-spglib_cell(lattice, atoms) = first(spglib_cell_atommapping_(lattice, atoms))
+spglib_atoms(atoms) = first(spglib_atommapping(atoms))
 
 
-@timing function spglib_get_symmetry(lattice, atoms; tol_symmetry=1e-5)
-    spglib = pyimport("spglib")
+@timing function spglib_get_symmetry(lattice, atoms; tol_symmetry=1e-5, max_symops=384)
```

Comment:
This `max_symops` is pretty much a magic number, so it should be documented where it comes from. If I recall correctly this is used for some pointer magic inside spglib, so if you get this wrong DFTK could segfault, which we definitely don't want. Is there no way to ask the library for this? I mean it might change in future versions or not?

Reply:
If it's internal it shouldn't be a kwarg and just set inside this file. Eg `const SPGLIB_MAX_SYMOPS` at the beginning of the file

Reply:
It is because one needs to pass in preallocated arrays to store the translations and rotations, it's indeed to force spglib not to try and fit in more symmetry ops than allowed. There's `get_multiplicity` actually to figure that out. I'll use that!